openwisp_controller/config/views.py

```diff
@@ -65,43 +79,40 @@ def get_relevant_templates(request, organization_id):
     queryset = (
         Template.objects.filter(**filter_options)
         .filter(org_filters)
+        .order_by("-required", "-default")
         .only("id", "name", "backend", "default", "required")
     )
     selected_templates = []
-    if device_id:
-        selected_templates = (
-            Config.objects.prefetch_related("templates")
-            .only("templates")
-            .get(device_id=device_id)
-            .templates.all()
-        )
-    if group_id:
-        selected_templates = (
-            DeviceGroup.objects.prefetch_related("templates")
-            .only("templates")
-            .get(pk=group_id)
-            .templates.filter(organization_id=organization_id)
-        )
+    if config_id:
+        try:
+            selected_templates = (
+                Config.objects.prefetch_related("templates")
+                .only("templates")
+                .get(pk=config_id)
+                .templates.filter(org_filters)
+                .filter(**filter_options)
```

Comment:
I spotted something that is worrying me: we aren't enforcing/asserting that this config is really assigned to a device which is related to `organization_id`. In theory it would be possible to call an `organization_id` which is valid and a `config_id` which shouldn't be allowed.

So I think filter options should include `device__organization_id=organization_id`, what do you think?

Watch out from query count increase due to this.